app/modules/asset_groups/resources.py

```diff
@@ -461,6 +461,32 @@ def patch(self, args, asset_group_sighting):
         AuditLog.patch_object(log, asset_group_sighting, args, duration=timer.elapsed())
         return asset_group_sighting
 
+    @api.permission_required(
+        permissions.ObjectAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'obj': kwargs['asset_group_sighting'],
+            'action': AccessOperation.DELETE,
+        },
+    )
+    @api.login_required(oauth_scopes=['asset_group_sightings:write'])
+    @api.response(code=HTTPStatus.CONFLICT)
+    @api.response(code=HTTPStatus.NO_CONTENT)
+    def delete(self, asset_group_sighting):
+        """Delete an asset group sighting by ID."""
+        asset_group = asset_group_sighting.asset_group
+        if [asset_group_sighting] == asset_group.asset_group_sightings:
+            # Delete the only asset group sighting deletes the asset group
+            try:
+                asset_group.delete()
+            except HoustonException as ex:
+                abort(
+                    ex.status_code,
+                    ex.message,
+                    edm_status_code=ex.get_val('edm_status_code', None),
+                )
+        else:
+            asset_group_sighting.delete()
```

Comment:
This is where this change gets tricky. 
Scenario
Asset group has AGS 1 & 2
AGS 1 has Assets W & X
AGS 2 has Assets Y & Z

These assets may or may not have annots hanging off them. 
If the entire asset group is deleted all those assets and annotations are deleted.

In this case, if we're deleting AGS 2 then the Assets Y & Z and any annotations hanging off them are now unreferenced. 

I think this actually need to be a call to a new asset_group.delete_asset_group_sighting(asset_group_sighting) 
function which first checks the asset references in the AGS and for any assets that are referenced by no other AGS, do a delete_cascade on the assets, only then delete the AGS.

Reply:
Thanks for looking into this!

Does this look right?

```diff
diff --git a/app/modules/asset_groups/models.py b/app/modules/asset_groups/models.py
index 627b4cff..0ae9703a 100644
--- a/app/modules/asset_groups/models.py
+++ b/app/modules/asset_groups/models.py
@@ -1642,6 +1642,14 @@ class AssetGroup(db.Model, HoustonModel):
         # for its completion)
         delete_remote.delay(str(self.guid))

+    def delete_asset_group_sighting(self, asset_group_sighting):
+        with db.session.begin(subtransactions=True):
+            for asset in self.assets:
+                asset_ags = self.get_asset_group_sightings_for_asset(asset)
+                if asset_ags == [asset_group_sighting]:
+                    asset.delete_cascade()
+            asset_group_sighting.delete()
+
     # stub of DEX-220 ... to be continued
     def justify_existence(self):
         if self.assets:  # we have assets, so we live on
```

Reply:
Looking at the slack conversation, should we limit the deletion of asset group sightings to those without sightings?

Reply:
Yes this looks right.
AGS both with and without Sightings need to be deleted, in fact without sightings is probably the most common. With sighting would be an exception with the frontend as it is right now. 
But of course design principles mean that we don't rely on that. This implementation should cover all options safely

Reply:
> With sighting would be an exception with the frontend as it is right now.
Agreed.
>But of course design principles mean that we don't rely on that. This implementation should cover all options safely
Agreed.